### DISCUSSION TITLE
Working with photos from SD card

### DISCUSSION THREAD

**Discussion by hugowaals** [Bodmer/TFT_eSPI#1986] [General]
Good day,

Is there a good example available in which photos from an SD card are displayed on a TFT display with an ILI9341 driver (320 x 240). Do the photos to be displayed have to be edited before they are displayed?

thank you for the assistance,
Hugo

**Comment by Bodmer** [comment]
There is an example here:
https://github.com/Bodmer/TFT_eSPI/tree/master/examples/Generic/ESP32_SDcard_jpeg

Note that to test this I used a seperate SD card interface module to the one on the display. The one on the dislay often has series resistors in the clock and data lines which slows the signals too much to work properly. First check the SD card library examples work OK, to test the card interface.

**Comment by hugowaals** [comment]
Dear Bodmer,

 

I've tried to test the SD card library as you suggested, but I'm getting an error code that I can't identify, see below. Also when I try the sample program for testing the SPI I get this error code.

 

I get this error both in the Arduino IDE (1.8.13) and with Visual Studio

 

I used this library: https://github.com/espressif/arduino-esp32

 

Error code:

 

Compiling 'SD_Test' for 'ESP32 Dev Module'

SPI.cpp: In member function void SPIClass::transferBytes(const uint8_t*, uint8_t*, uint32_t)

 

SPI.cpp: 297:43: error: invalid conversion from 'const uint8_t* {aka const unsigned char*}' to 'uint8_t* {aka unsigned char*}' [-fpermissive]

   spiTransferBytes(_spi, data, out, size)

Error compiling libraries

Build failed for project 'SD_Test'

 

SPI.h:26: In file included from

SPI.cpp:22: from

esp32-hal-spi.h:108: note    initializing argument 2 of void spiTransferBytes(spi_t*, uint8_t*, uint8_t*, uint32_t)

   void spiTransferBytes(spi_t * spi, uint8_t * data, uint8_t * out, uint32_t size)

 

I hope you can help me out!

 

Thanks in advance,

Hugo

 

Van: Bodmer ***@***.***>
Beantwoorden - Aan: Bodmer/TFT_eSPI ***@***.***>
Datum: vrijdag 2 september 2022 om 22:49
Aan: Bodmer/TFT_eSPI ***@***.***>
CC: "Hugo (Prive)" ***@***.***>, Author ***@***.***>
Onderwerp: Re: [Bodmer/TFT_eSPI] Working with photos from SD card (Discussion #1986)

 

There is an example here:
https://github.com/Bodmer/TFT_eSPI/tree/master/examples/Generic/ESP32_SDcard_jpeg

Note that to test this I used a seperate SD card interface module to the one on the display. The one on the dislay often has series resistors in the clock and data lines which slows the signals too much to work properly. First check the SD card library examples work OK, to test the card interface.
